ethereum/client/src/lib.rs

```diff
@@ -358,72 +430,189 @@ impl Contracts {
 
 #[cfg(test)]
 mod tests {
-    use super::*;
-    use alloy_node_bindings::{Anvil, AnvilInstance};
-    use alloy_primitives::keccak256;
-    use alloy_provider::HttpProvider;
-    use alloy_transport_http::Http;
+
+    use alloy::hex;
+    use alloy::node_bindings::{Anvil, AnvilInstance};
+    use alloy::primitives::keccak256;
+    use alloy::providers::ProviderBuilder;
+    use alloy::transports::http::{Client, Http};
     use binary_merkle_tree::{merkle_proof, merkle_root, verify_proof, Leaf, MerkleProof};
     use primitive_types::H256;
     use sp_core::KeccakHasher;
 
-    #[allow(unused, unreachable_pub)]
-    pub fn spawn_anvil() -> (HttpProvider<Ethereum>, AnvilInstance) {
-        spawn_anvil_with(std::convert::identity)
-    }
+    use crate::abi::{
+        ERC20Mock, ERC20Treasury, MessageQueue, ProxyContract, Relayer, Verifier, VerifierMock,
+    };
 
-    #[allow(unused, unreachable_pub)]
-    pub fn spawn_anvil_with(
-        f: impl FnOnce(Anvil) -> Anvil,
-    ) -> (HttpProvider<Ethereum>, AnvilInstance) {
-        let anvil = f(Anvil::new()).try_spawn().expect("could not spawn anvil");
-        (anvil_http_provider(&anvil), anvil)
-    }
+    use super::*;
 
-    #[allow(unused, unreachable_pub)]
-    pub fn anvil_http_provider(anvil: &AnvilInstance) -> HttpProvider<Ethereum> {
-        http_provider(&anvil.endpoint())
+    fn build_contracts<P, T>(provider: P) -> Result<Contracts<P, T, Ethereum>, Error>
```

Comment:
Why not to take addresses from `DeploymentEnv` after `deploy()`?

Reply:
Done